zap/zap.gradle.kts

```diff
@@ -34,8 +34,13 @@ java {
             languageVersion.set(JavaLanguageVersion.of(8))
         }
     } else {
-        sourceCompatibility = JavaVersion.VERSION_1_8
-        targetCompatibility = JavaVersion.VERSION_1_8
+        if (System.getProperty("os.arch") == "aarch64") {
+            sourceCompatibility = JavaVersion.VERSION_11
+            targetCompatibility = JavaVersion.VERSION_11
+        } else {
+            sourceCompatibility = JavaVersion.VERSION_1_8
+            targetCompatibility = JavaVersion.VERSION_1_8
+        }
```

Comment:
This should be done in other pull request.

Reply:
But is this really needed? It's possible to compile targetting older versions with newer ones.

Reply:
When setting up a new dev environment and running `./gradlew copyWeeklyAddOns`

I kept getting this error...

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':addOns:accessControl:jarZapAddOn'.
> Failed to calculate the value of task ':addOns:accessControl:compileJava' property 'javaCompiler'.
   > Unable to download toolchain matching these requirements: {languageVersion=8, vendor=any, implementation=vendor-specific}
      > Unable to download toolchain. This might indicate that the combination (version, architecture, release/early access, ...) for the requested JDK is not available.
         > Could not read 'https://api.adoptopenjdk.net/v3/binary/latest/8/ga/mac/aarch64/jdk/hotspot/normal/adoptopenjdk' as it does not exist.
``` 

The above error is actually being produced because of this https://github.com/zaproxy/zap-extensions/blob/9bf0285550a110bb8dc03bb7be7b219d16dca6ca/addOns/addOns.gradle.kts#L97

And I have this update in a local zap-extensions branch to address it. 
```
if (System.getenv("ZAP_RELEASE") != null && System.getProperty("os.arch") != "aarch64") {
            toolchain {
                languageVersion.set(JavaLanguageVersion.of(8))
            }
        }
```         

I was just adding this as the jdk8 M1 support is pretty bad in general and thought this might make getting started easier.

I'll take it out of this PR  though.

Reply:
Are you setting the `ZAP_RELEASE` env var?

Reply:
I can make a PR for this task to make it more flexible. Perhaps a new task for local dev, `copyWeeklyAddonsDev` that doesn't set `ZAP_RELEASE`  🤷‍♂️

Reply:
It's not necessary a new task, we should change CI to set the var when releasing.